src/App.tsx

```diff
@@ -1,74 +1,221 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useCallback, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { UpdateTodoData } from './types/types';
 
-const USER_ID = 0;
+const USER_ID = 10917;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [status, setStatus] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    let updatedTodos;
+
+    switch (status) {
+      case 'Active': {
+        updatedTodos = visibleTodos.filter(todo => !todo.completed);
+        break;
+      }
+
+      case 'Completed': {
+        updatedTodos = visibleTodos.filter(todo => todo.completed);
+        break;
+      }
+
+      default: updatedTodos = visibleTodos;
+    }
+
+    setVisibleTodos(updatedTodos);
+  }, [todos, setVisibleTodos]);
+
+  const addTodo = useCallback(async (title: string) => {
+    try {
+      const newTodo: Todo = await client.post(`/todos?userId=${USER_ID}`, {
+        userId: USER_ID,
+        completed: false,
+        title,
+      });
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+      setErrorMessage('');
+
+      return newTodo;
+    } catch (error) {
+      setErrorMessage('Unable to add a todo');
+      throw new Error('Error');
+    }
+  }, []);
+
+  const updateTodo = async (todoId: number, data: UpdateTodoData) => {
+    try {
+      await client.patch(`/todos/${todoId}`, data);
+
+      setTodos((prevTodos: Todo[]) => prevTodos.map((todo: Todo): Todo => {
+        return todo;
+      }));
+      setErrorMessage('');
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+      throw new Error('Error');
+    }
+  };
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      const updatedTodo = await client.delete(`/todos/${todoId}`);
+
+      setTodos((prevTodos: Todo[]) => prevTodos.map((todo: Todo): Todo => {
+        if (todo.id === todoId) {
+          return updatedTodo as Todo;
+        }
+
+        return todo;
+      }));
+      setErrorMessage('');
+
+      return updatedTodo;
+    } catch (error) {
+      setErrorMessage('Unable to delete a todo');
+      throw new Error('Error');
+    }
+  };
```

Comment:
Would be nice to have these functions in a separate file

Reply:
sure